src/main/scala/rocket/RocketCore.scala

```diff
@@ -702,6 +702,13 @@ class Rocket(implicit p: Parameters) extends CoreModule()(p)
          wb_reg_cinst, wb_reg_cinst)
   }
 
+  val max_core_cycles = PlusArg("max-core-cycles",
+    default = 0,
+    docstring = "Maximum Core Clock cycles simulation may run before timeout. Ignored if 0 (Default).")
+  when (max_core_cycles > UInt(0)) {
+    assert (csr.io.time < max_core_cycles, "Maximum Core Cycles reached.")
+  }
```

Comment:
Yeah, it's correct as-is.  I was just saying that the when is unnecessary, since the when is only false when max_core_cycles is 0, in which case the assertion is always false, too.